web/src/portico/signup.ts

```diff
@@ -24,10 +25,18 @@ $(() => {
         // was just reloaded due to a validation failure on the backend.
         password_quality($password_field.val()!, $("#pw_strength .bar"), $password_field);
 
+        const debounced_password_quality = _.debounce((password_value: string, $field: JQuery) => {
+            password_quality(password_value, $("#pw_strength .bar"), $field);
+        }, 300);
+
         $password_field.on("input", function () {
-            // Update the password strength bar even if we aren't validating
-            // the field yet.
-            password_quality($(this).val()!, $("#pw_strength .bar"), $(this));
+            const password_value = $(this).val()!;
+
+            if (password_value.length > 30) {
+                debounced_password_quality(password_value, $(this));
+            } else {
+                password_quality(password_value, $("#pw_strength .bar"), $(this));
+            }
```

Comment:
This has a bug: when quickly backspacing a password from 32 to 31 to 30 characters, the 31-character update will be delayed until after the 30-character update. This is what `_.debounce(…).cancel()` is for (always read the [documentation](https://lodash.com/docs/4.17.15#debounce)!).